doc/specs/vulkan/chapters/VK_KHR_swapchain/wsi.txt

```diff
@@ -454,14 +455,16 @@ endif::VK_VERSION_1_1,VK_KHR_device_group[]
 The pname:surface must: not be destroyed until after the swapchain is
 destroyed.
 
-If pname:oldSwapchain is not dlink:VK_NULL_HANDLE then pname:surface must:
-be associated with pname:oldSwapchain.
-Otherwise, the native window referred to by pname:surface must: not already
-be associated with another swapchain, and must: not already be associated
-with a non-Vulkan graphics API surface.
+If pname:oldSwapchain is dlink:VK_NULL_HANDLE, the swapchain cannot: be
```

Comment:
I just noticed your use of "cannot:" here and elsewhere.  It's consistent with the old text at the top of the file, but I think it wrong (or at least isn't the best) for what the intention of "cannot:" is (i.e. describing what an application cannot do).  The old language that uses "must:" (i.e. "... must: not already
-be associated with another swapchain ...") seems better.  Please fix that.

Reply:
Seems to fit in this case. The creation will fail if the app attempts to do so -> "cannot".
In contrast to "must", which would mean doing so is undefined behavior.

Reply:
I think the problem I'm having is with how you are placing the nouns and verbs (I'm not an English major, so I don't know the correct terms to describe what I'm trying to say, but there's some concepts around the subjects of verbs).  Your wording makes it sound like the swapchain is the subject of the "cannot:" (i.e. a swapchain is part of the implementation; for which "cannot:" cannot be used) instead of the application.  If you say the application cannot: do something I wouldn't object.

At this point, I prefer to keep the old wording.

Reply:
I don't think either is correct.  "cannot:" (normative form) is supposed to be reserved for statements like "The application cannot: destroy physical devices," because there is actually no API to destroy a physical device.  It never refers to something affecting the implementation, which is arguably the case here.  Hence, it's used pretty sparsely in the spec, because there are few things Vulkan prevents applications from attempting at all ;-)

"must: not" is wrong for the reason @krOoze notes.

What we're really supposed to do, based on examples elsewhere in the spec, is refer to the error code generated.  I.e., "If pname:oldSwapchain is dlink:VK_NULL_HANDLE and the native window referred to by pname:surface is already associated with a swapchain, VK_ERROR_NATIVE_SURFACE_IN_USE_KHR must: be returned."

Reply:
That's a shame. **cannot** seem to be born for this situation;
**cannot** definition:

> This word means that the API and/or the execution environment provide no mechanism through which the application can express or accomplish the action described.

The app can "express" it but won't "accomplish" creating swapchain with surface already in use.

@cubanismo I somewhat dislike that style, because it is incareful with the RFC keyword, which is to be always absolute; TBH it lies. E.g. to me `VK_ERROR_OUT_OF_MEMORY` seems as much valid in the situation as `VK_ERROR_NATIVE_SURFACE_IN_USE_KHR`, but that is out of the question if one of them says "this one **must** be returned".

I am gonna change it that way, because as you say it seems to be the established style, but it is perhaps worth an Issue...

Reply:
updated...